src/playercards/AllCardsBag.ttslua

```diff
@@ -249,34 +562,80 @@ function getCardById(params)
   return cardIdIndex[params.id]
 end
 
--- Returns a list of cards from the bag matching a class and level (0 or upgraded)
--- Params table:
---     class: String class to retrieve ("Guardian", "Seeker", etc)
---     isUpgraded: true for upgraded cards (Level 1-5), false for Level 0
--- Return: If the indexes are still being constructed, returns an empty table.
---     Otherwise, a list of tables, each with the following fields
---       cardData: TTS object data, suitable for spawning the card
---       cardMetadata: Table of parsed metadata
-function getCardsByClassAndLevel(params)
+-- Used to index into the supplemental card indices 
+---@param indexName exact string of the index variable to be referenced, minus the 'Index' part
+---@param includeOthers boolean of whether or not to include non-player cards
+---@param value key used to index into the dictionary
+---@return List of cards located at the precise index given the provided parameters
+function getCardsByData(params)
+  -- Make sure indices are finished building
   if (not indexingDone) then
     broadcastToAll("Still loading player cards, please try again in a few seconds", {0.9, 0.2, 0.2})
     return { }
   end
-  local upgradeKey
-  if (params.upgraded) then
-    upgradeKey = "-upgrade"
-  else
-    upgradeKey = "-level0"
+  -- Clean up string input
+  local value = nil
```

Comment:
I think `tableKey` would be a better name than `value` considering it can be a string or number

Reply:
100% agree, changed!